### DISCUSSION TITLE
tag syntax

### DISCUSSION THREAD

**Discussion by imcineri** [vkbo/novelWriter#658] [Help]
Could someone tag the following sentence for me please? I don't get how to set them up.

Bob and Joe were at the bar Friday night.

Thank you in advance.

**Comment by vkbo** [comment]
```
### Scene

@char: Bob, Joe

Bob and Joe were at the bar Friday night.
```
https://novelwriter.readthedocs.io/en/stable/usage_structure.html#tag-references

Hope that helps :)

**Reply by imcineri** [reply]
Thank you for your reply. Sort of helps. I had first thought the tags were inline. Now I see they are set at the top of the "chapter" for what is to follow in the text. I will re-read the relevant docs with that in mind now. I may have further questions. Thanks again.

**Reply by imcineri** [reply]
Well, after a reasonable attempt to get this, I'll be scheduling an MRI next week. Would there happen to be a dummy sample that correctly uses all the features of this program that I can use to follow along with as I, once again, try to get this dang prog? Or maybe a different doc that's been dumbed down?

**Reply by vkbo** [reply]
You can create a new project from the "Open Project" dialog and populate it with example files. The example project uses all the features to demonstrate them. It's what I use for testing when working on the code.

Is it the tags and references you have problems with?

**Reply by vkbo** [reply]
As the person who came up with many of the concepts and wrote the code, it is difficult for me to judge how easy the documentation is to read. Most of my co-conspirators when it comes to developing the features have also been with the project since early on, and I think most of us lack the new-to-this perspective.

I'm happy to take suggestions on how to improve the documentation. Maybe it would be useful with a brief introductory summary of the functionality? There used to be more information in the main README, but I've cut down a lot on it as the documentation has grown.

**Reply by imcineri** [reply]
That is exactly what I'm looking for. Thank you. As to your doc comment, I had wondered if that was the case. I also thought maybe I had chosen above my level. Is your intended market new writers or broader?

**Reply by vkbo** [reply]
To be honest, the original "marked" was me and one of my colleagues at work and not much more. We are both scientists and used to writing technical and scientific documents in LaTeX using code editors, and we connected over both writing fiction as a hobby. As a result, the original design was definitely geared towards people used to working with programming tools more than WYSIWYG style editors. Very few of my colleagues ever touched office applications at work and most of us work on Linux and thus can't use many of the commercial tools out there.

The main inspiration for the design are a mix of features from yWriter (outlining novels), Zim-wiki (note-taking), Visual Studio Code (plain text editor) and GhostWriter (markdown editor). The current program has evolved somewhat, but is still focused on writing plain text files extended with useful tools and export formats geared towards writing fiction.

I'm definitely happy that other people find it useful, but the app is clearly more appealing to people with a similar preference in writing tools. The tool set has been extended based on feedback from writers with much more experience than me. I've mostly done non-fiction freelance writing and scientific writing. Writing fiction is still a hobby at the moment, although I'm actively learning the trade. I intend to keep working on novelWriter as the main app for my own writing. I find that I quite enjoy both sides of that. :)

**Reply by imcineri** [reply]
It's a lovely project with maybe more potential than you've considered. I have a thought or two about your docs but I'd like to refrain for now until I have messed around with the prog. Unless I've missed the feature, I will say I believe you should immediately implement a "delete entire project" option which goes so far as to remove the project from the list on the "open project" pop-up. I mean wipe the whole thing. I don't code so I don't know but patch and release today if easy. For now I will just uninst/reinst and start again. The dummy doc will be a great help. Thanks.

**Reply by vkbo** [reply]
Oh, if you mean removing the example project, I expect people to just delete the project folder where they put it. The remove button only removes the entry in the list as you say. I think deleting projects from inside the app is unnecessary. Most apps don't do that.

I'm not sure what you mean with uninstall/reinstall? That does nothing to the projects.

**Reply by imcineri** [reply]
So to delete a project you have to dele from the bottom up and can't dele trash folder so project is never gone from the list in the "open project" pop up. I just uninstalled through control panel and manually dele project folders, emptied my trash, restarted my pc, reinstalled novelwriter and, when I opened it, the popup was populated with the deleted projects names. Should have been empty. Right?

**Reply by vkbo** [reply]
Ah, the list in "Open Projects" is just a record of your last opened projects. It doesn't represent what exists on your harddrive. novelWriter doesn't scan your disks. It only knows about projects you've opened before. Removing them from the list just removes that entry.

**Reply by imcineri** [reply]
And how do I remove them from the list?

**Reply by imcineri** [reply]
Okay. I got what you meant about the list. I may have a bug for you. When I attempt to resize the Preferences dialogue box it crashes novelWriter. Reproduced several times. It seems to be possessed in some way.

**Reply by vkbo** [reply]
Hmm. It works fine here. Please post the content of the crash dialog box if it appears. It should have all the debug information.

**Reply by imcineri** [reply]
No dialogue box but the entire window becomes unclickable. I shut it down with Task Manager. Thing is, I'm doing because the Pref dialogue oversizes my screen. Can't finalize changes. I'll check my Windows settings for my display rez and etc.

**Reply by vkbo** [reply]
That is weird. The Python code doesn't capture resize events, so it may be an issue on the Qt side of things.

**Reply by imcineri** [reply]
I have found that if I open the Pref Dialogue and then go to my Windows Display scaling feature and change it I can make the Okay/Cancel buttons on the bottom visible but if I change scaling before running novelWriter I just have my original bug of the box being too tall for the window. Using that scaling trick I changed the fonts back to lower pixels in the Pref dialogue and now all is well. It was my bad but users with vision issues may have a problem. Looks like they will be able to go in and enlarge their program fonts but if they need to change prefs after that they may not have access the the bottom buttons. idk

**Reply by vkbo** [reply]
Do you use GUI scaling that isn't 100% or 200%?

**Reply by imcineri** [reply]
My GUI was at 125 and that will eliminate the bottom of the dialogue.

**Reply by vkbo** [reply]
Qt has very poor support for fractional GUI scales. My code takes it into account, but I can't do much about the Qt side of it unfortunately. I think Qt 6 supports it properly, but I'm not moving novelWriter to Qt 6 for some time yet.

**Reply by imcineri** [reply]
I don't think a normal person would ever notice it. I'm pretty weird.

**Reply by imcineri** [reply]
I've been working on a regular persons guide over the last few days and it's really helped me learn your program. If you would like to take a look at it let me know.

**Reply by lcarsos** [reply]
Sorry to necropost, but I just discovered novelWriter today, and also just spent about 2 hours trying to figure out how the hell to get tags to work.

First off, thank you for the suggestion of creating the example project. In my poking around I didn't think that there actually was an example project, just some default folders that get created.

For me the thing that tripped me up is that nowhere in the documentation, or in the README, or in the error when a tag doesn't work is the text:

> To create a tag, create an Item in the folder for the tag type, and in the text of the item must be `@tag: tagname`. `@tag` is literal, it is not the type of the tag.

**Reply by Ellipsine** [reply]
Hi @lcarsos 
For a tag to work, you need the file which defines the tag, and to tell the software that this file is the tag file you wish.

Let's take a pratical example. In my novel, I want a POV from "Bart" character. So I write "@POV: Bart" on my novel part, which is the good syntax. 

But the software does not find "Bart" as a character. So I create "Bart" file in the "CHARACTERS" part. In my "Bart" file I need to precise to the software that this file is also the "Bart" tag. So I tell it : "@tag: Bart". As this file is in the CHARACTER section created by the software, it knows Bart is a character. 

In the end, I have "Bart" in my characters files with the precision "@tag: Bart" inside it, and "@POV: Bart" in my novel files. 

I hope it helps and answers your question.

**Reply by vkbo** [reply]
Hi @lcarsos,

The relevant section in the documentation is https://novelwriter.readthedocs.io/en/latest/project_notes.html

I am very happy for suggestions on how to improve it or to make it clearer. Perhaps I should add an example.

I also have an issue #823 up to improve the user feedback in the editor on tags and references.

**Reply by mogplus88** [reply]
I got a 404 error when I clicked on the link to readthedocs. Has it moved?

**Reply by vkbo** [reply]
Yes, a long time ago. That page doesn't exist either, but tags and references are covered here: https://novelwriter.io/docs/project_references.html
